_posts/2019-09-23-exposing-http-bridge.md

```diff
@@ -0,0 +1,270 @@
+
+```shell
+curl -X POST \
+  http://my-bridge.io/topics/my-topic \
+  -H 'content-type: application/vnd.kafka.json.v2+json' \
+  -d '{
+    "records": [
+        {
+            "key": "key-1",
+            "value": "value-1"
+        },
+        {
+            "key": "key-2",
+            "value": "value-2"
+        }
+    ]
+}'
+```
+
+After writing the messages into the topic, the bridge replies with an HTTP status code `200 OK` and a JSON paylod describing in which partition and at which offset the messages are written.
+In this case, the auto created topic has just one partition and the reply can look like the following one.
+
+```json
+{ 
+   "offsets":[ 
+      { 
+         "partition":0,
+         "offset":0
+      },
+      { 
+         "partition":0,
+         "offset":1
+      }
+   ]
+}
+```
+
+# Consuming messages
+
+Consuming messages is not so simple as producing because there are several steps to do which involve different endpoints.
+First of all, creating a consumer through the `/consumers/{groupid}` endpoint doing an HTTP POST with a body containing some of the supported configuration parameters, the name of the consumer and the data format (JSON or binary).
+In the following snippet, we are going to create a consumer named `my-consumer` and joining the consumer group `my-group`.
+
+```shell
+curl -X POST http://my-bridge.io/consumers/my-group \
+  -H 'content-type: application/vnd.kafka.v2+json' \
+  -d '{
+    "name": "my-consumer",
+    "format": "json",
+    "auto.offset.reset": "earliest",
+    "enable.auto.commit": false
+  }'
+```
+
+After creating a corresponding native Kafka consumer connected to the Kafka cluster, the bridge replies with an HTTP status code `200 OK` and a JSON payload containing the URI that the HTTP client has to use in order to interact with such a consumer for subscribing and receiving messages from topics.
+
+```json
+{ 
+   "instance_id":"my-consumer",
+   "base_uri":"http://my-bridge.io:80/consumers/my-group/instances/my-consumer"
+}
+```
+
+# Subscribing to the topic
+
+The most used way for a Kafka consumer to get messages from a topic is to subscribe to that topic as part of a consumer group and getting partitions assigned automatically.
+Using the HTTP bridge, it's possible through an HTTP POST to the `/consumers/{groupid}/instances/{name}/subscription` endpoint providing in a JSON formatted payload the list of topics to subscribe to or a topics pattern.
+With the following snippet, the consumer is subscribing to the `my-topic` topic.
+
+```shell
+curl -X POST http://my-bridge.io/consumers/my-group/instances/my-consumer/subscription \
+  -H 'content-type: application/vnd.kafka.v2+json' \
+  -d '{
+    "topics": [
+        "my-topic"
+    ]
+}'
+```
+
+In this case, the bridge just replies with an HTTP status code `200 OK` with an empty body.
+
+# Consuming messages
```

Comment:
You used auto-commit set to false. So I think you need to show how to commit the offsets (which you probably should show and explain anyway).